docker/docker-entrypoint.sh

```diff
@@ -1,47 +1,172 @@
 #!/bin/bash
 set -eo pipefail
 
-# mysql_log logs messages with a timestamp and optional color formatting.
+# mysql_log prints a structured log line with optional color formatting
+#
 # Arguments:
-#   $1 - Log type (e.g., Note, Warn, ERROR)
-#   $@ - Log message (if empty, reads from stdin)
+#   $1            - Timestamp (empty string "" for current time)
+#   $2..$n-1      - Metadata fields (last argument before optional type index is the message)
+#   $n (optional) - Numeric index of metadata field to use for determining log type/color
+#
+# Description:
+#   Prints a structured log line with timestamp, metadata, and message.
+#   Colors the output based on log type:
+#     Warning - yellow
+#     ERROR   - red
+#     Debug   - blue
+#   The log type is determined from the metadata field at type_index (default 0).
+#   The message is always the second-to-last argument.
```

Comment:
This isn't actually true, right?

Reply:
It is but it's also a result of me trying to match the MySQL logs. I'll revert this one.